### PR TITLE
Port frost-ristretto255 to frost-core

### PR DESCRIPTION
**Pull request by dconnolly**

### CODE REVIEW

**Comment by dconnolly** [comment]
@conradoplg all tests and checks passing now, want to take a look?

**Comment by dconnolly** [comment]
> @conradoplg all tests and checks passing now, want to take a look?

I think I'm going to start moving the generic functions to generic associated trait functions / methods in a new PR based on this branch

**Comment by dconnolly** [comment]
> > @conradoplg all tests and checks passing now, want to take a look?
> 
> I think I'm going to start moving the generic functions to generic associated trait functions / methods in a new PR based on this branch

Actually I need to move the test vector tests up too

File: frost-ristretto255/tests/proptests.rs [hunk 1]
```diff
@@ -40,36 +37,32 @@ impl SignatureCase {
...
+        // // The signature data is stored in (refined) byte types, but do a round trip
+        // // conversion to raw bytes to exercise those code paths.
+        // let sig = {
+        //     let bytes: [u8; 64] = self.sig.into();
+        //     Signature::<C>::from_bytes(bytes)
+        // };
+
+        // // Check that the verification key is a valid key.
+        // let pub_key = VerifyingKey::<C>::from_bytes(pk_bytes)
+        //     .expect("The test verification key to be well-formed.");
```
Comment (conradoplg) [inline frost-ristretto255/tests/proptests.rs:62]: Were these commented out on purpose? If yes, I suggest adding the reason as comment, or just deleting it

**Review by conradoplg — DISMISSED** [review]
Looks great! Just one optional comment

File: frost-ristretto255/tests/proptests.rs (hunk 1 above)
Comment (dconnolly) [reply frost-ristretto255/tests/proptests.rs:62]: Oh yeah I forgot this, I'll fixup

Comment (dconnolly) [reply frost-ristretto255/tests/proptests.rs]: Resolved in d849c281b6a89f74bbe8e96ecc518f9a7c49698d

**Comment by dconnolly** [comment]
I also enabled the codecov comment like we have with Zebra.

image

image


image

image

**Review by conradoplg — APPROVED** [review]
Looks good

**Comment by dconnolly** [comment]
@mergify update

**Pull request merged**